### DISCUSSION TITLE
Maintaining data after migration from UWP to Desktop

### DISCUSSION THREAD

**Discussion by dajbych** [microsoft/WindowsAppSDK#1689] [Q&A] [answered]
My UWP app is distributed through the Microsoft Store and its local data are stored in the `C:\Users\<username>\AppData\Local\Packages\<package family name>\` folder. UWP will not hot have .NET 6 support. It is recommended to migrate to WinUI 3/Windows App SDK/Win32 (Desktop). When the migrated app is distributed via Microsoft Store as a higher version of a previous UWP app, will it still have access to this directory with the same data?

**Comment by andrewleader** [comment, marked as answer]
Hi @dajbych, in short - yes, your app will still have access to the same data and directory and be able to seamlessly upgrade data from UWP to WinUI 3 Desktop.

When you migrate to WinUI 3, you will want to make sure you still build a packaged MSIX app, and make sure that the **identity** of your app is the same (identity is specified in the manifest). As long as you have the same identity, your WinUI 3 version of your app will be able to access the exact same data your UWP app used to - simply keep using the same ApplicationData APIs!
